specification.markdown

```diff
@@ -773,7 +779,11 @@ description: The version of the DocTypeExtension. Different DocTypeExtensionVers
 
 ## Global Elements
 
-EBML defines these Global Elements which MAY be stored within any Master Element of an EBML Document as defined by their Element Path.
+EBML allows some special Elements to be found within more than one parent in an EBML Document. These Elements are called Global Elements. There are 2 Global Elements that can be found in any EBML Document: the CRC-32 Element and the Void Element. An EBML Schema MAY add other Global Elements to the format it defines.
```

Comment:
I think it should be allowed and that's how the `Void` path reads (it took me a while to figure it out back).

`Void` can be used to void some elements, including a Segment. Unlike CRC-32 it doesn't have any relation to a parent nor does it need to.

libebml was originally designed to handle such case, it should be able to handle Void without a parent.

Reply:
I am in favor of allowing `Void` at the top level, it may be used in some corner cases and does not hurt.
Actually `CRC-32` definition could be slightly changed for also being at the top level, from first element after EBML header and up to end of file, so I prefer the new wording.

Reply:
> then concatenate them as an EBML Stream, then ?_?

You get the point. Up to end of file or next `EBML` element.

But before debating about `CRC-32`, let's agree first on `Void` element.

Reply:
> Why allow this? It allows a file where the header is exabytes into the file.

`EBML` element must be first IIRC, so still no `Void` element first.

> Why allow such a mess?

In Matroska `Segment` must be unique. But in case it is used elsewhere, a top element could be several times, and "voiding" one can permit to avoid to rewrite the whole file while permitting to remove nicely an top element.

> SeekHead

Is from Matroska, not EBML. I get that the goal is Matroska, but I am not convinced that we should force that in EBML itself. Maybe forbidding `Void` at top level in Matroska specs?

Reply:
> I analyzed over 100,000 mkv files in archive.org and never see a Void at top-level.

Where you looking for mkv files where the `Segment` was `Void` and thus the file was likely empty/unusable ? I seems logical that archive.org would not have such a file. That doesn't mean anything to how `Void` should be interpreted.

Reply:
I agree with @JeromeMartinez here. It wouldn't make much sense for Matroska (we can discuss this further). But for EBML formats it could make sense.

For example if you have a format which is a big flat table of elements like this:
```
<EBML/>
<XEntry/>
<XEntry/>
<YEntry/>
<XEntry/>
<YEntry/>
<ZEntry/>
<XEntry/>
...
```

You could want to erase one of the entries without having to rewrite the whole file.